indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalInsertAction.java

```diff
@@ -253,11 +256,27 @@ public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox)
     for (DataSegment segment : retVal.getSegments()) {
       metricBuilder.setDimension(DruidMetrics.INTERVAL, segment.getInterval().toString());
       toolbox.getEmitter().emit(metricBuilder.build("segment/added/bytes", segment.getSize()));
+      // Emit the segment related metadata using the configured emitters
+      this.emitSegmentMetadata(segment, toolbox);
     }
 
     return retVal;
   }
 
+  private void emitSegmentMetadata(DataSegment segment, TaskActionToolbox toolbox)
+  {
+    SegmentMetadataEvent event = new SegmentMetadataEvent(
+        segment.getDataSource(),
+        DateTime.now(DateTimeZone.UTC),
+        segment.getInterval().getStart(),
+        segment.getInterval().getEnd(),
+        segment.getVersion(),
+        segment.getLastCompactionState() == null ? Boolean.FALSE : Boolean.TRUE
```

Comment:
if you follow @lqxshay 's suggestion of using a primitive boolean type, then you can avoid this complexity and replace this line with `segment.getLastCompactionState() != null`